src/System.Windows.Forms/src/System/Windows/Forms/Controls/Obsolete/ContextMenu/MenuItemCollection.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.ComponentModel;
+
+namespace System.Windows.Forms;
+
+public abstract partial class Menu
+{
+    [Obsolete(
+        Obsoletions.MenuItemCollectionMessage,
+        error: false,
+        DiagnosticId = Obsoletions.MenuItemCollectionDiagnosticId,
+        UrlFormat = Obsoletions.SharedUrlFormat),
+        EditorBrowsable(EditorBrowsableState.Never),
+        ListBindable(false)]
+    public class MenuItemCollection : IList
+    {
+        public MenuItemCollection(Menu owner) => throw new PlatformNotSupportedException();
+
+        public virtual MenuItem this[int index] => throw new PlatformNotSupportedException();
+
+        object IList.this[int index]
+        {
+            get => throw new PlatformNotSupportedException();
+#pragma warning disable CS8769 // Nullability of reference types in type of parameter doesn't match implemented member (possibly because of nullability attributes).
+            set => throw new PlatformNotSupportedException();
+#pragma warning restore CS8769 // Nullability of reference types in type of parameter doesn't match implemented member (possibly because of nullability attributes).
+        }
+
+        public virtual MenuItem this[string key] => throw new PlatformNotSupportedException();
+
+        public int Count => throw new PlatformNotSupportedException();
+
+        object ICollection.SyncRoot => throw new PlatformNotSupportedException();
+
+        bool ICollection.IsSynchronized => throw new PlatformNotSupportedException();
+
+        bool IList.IsFixedSize => throw new PlatformNotSupportedException();
+
+        public bool IsReadOnly => throw new PlatformNotSupportedException();
+
+        public virtual MenuItem Add(string caption) => throw new PlatformNotSupportedException();
+
+        public virtual MenuItem Add(string caption, EventHandler onClick) => throw new PlatformNotSupportedException();
+
+        public virtual MenuItem Add(string caption, MenuItem[] items) => throw new PlatformNotSupportedException();
+
+#pragma warning disable CS8767 // Nullability of reference types in type of parameter doesn't match implicitly implemented member (possibly because of nullability attributes).
+        public virtual int Add(object value) => throw new PlatformNotSupportedException();
+#pragma warning restore CS8767 // Nullability of reference types in type of parameter doesn't match implicitly implemented member (possibly because of nullability attributes).
```

Comment:
We had recently changed this pattern to have to explanation comment only at the suppression site, espessially when the disable and restore are so close to each other.
```suggestion
#pragma warning restore CS8767
```